getndazn/dazn-lambda-powertools
# feat(middleware): add alb as an event source

## What did you implement:
Add ALB as a supported event source for the correlation IDs middleware.
Closes #130 



## How did you implement it:
Created a new event source detector specifically for ALB. Also modified API Gateway to exclude any ELB events as both event payloads, although quite similar, do have their differences.

On API Gateway, the headers are case-sensitive, but on ALB the header names you get in the request object are all lower-cased. The change here introduces a new const for detecting user agent. The correlation ID key is still preserved as `User-Agent` to maintain consistency. The tests also cover this.


## How can we verify it:
Add a Lambda function behind an ALB target group to view correlations IDs being detected and forwarded on where required.

ALB event test added to verify.



## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
